profile/fs_interp.py

```diff
@@ -0,0 +1,77 @@
+import click
+import math
+import time
+import numpy as np
+import matplotlib.pyplot as plt
+from pyffs.func import dirichlet_fs
+from pyffs.interp import fs_interp
+
+
+@click.command()
+@click.option("--n_trials", type=int, default=50)
+def profile_fs_interp(n_trials):
+    print(f"\nCOMPARING FS_INTERP WITH {n_trials} TRIALS")
+
+    # parameters of signal
+    T, T_c, M = math.pi, math.e, 1000
+
+    # sweep over number of interpolation points
+    a, b = T_c + (T / 2) * np.r_[-1, 1]
+    n_std = 1.0
+    real_x = {"complex": False, "real": True}
+    N_FS_vals = [1001, 3001, 10001, 30001, 100001, 30001, 100001]
+    proc_time = dict()
+    proc_time_std = dict()
+    for N_FS in N_FS_vals:
+        print("\nNumber of FS coefficients : {}".format(N_FS))
+        proc_time[N_FS] = dict()
+        proc_time_std[N_FS] = dict()
+
+        # compute FS coefficients
+        diric_FS = dirichlet_fs(N_FS, T, T_c)
+
+        # Loop through functions
+        for _f in real_x:
+            timings = []
+            for _ in range(n_trials):
+                start_time = time.time()
+                fs_interp(diric_FS, T, a, b, M, real_x=real_x[_f])
+                timings.append(time.time() - start_time)
+            proc_time[N_FS][_f] = np.mean(timings)
+            proc_time_std[N_FS][_f] = np.std(timings)
+            print("{} version : {} seconds".format(_f, proc_time[N_FS][_f]))
+
+    # plot results
+    markers = ["o", "^", "v", "x", ">", "<", "D", "+"]
+    plt.figure()
+    for i, _f in enumerate(real_x):
+        _proc_time = []
+        _proc_time_std = []
+        for N_FS in N_FS_vals:
+            _proc_time.append(proc_time[N_FS][_f])
+            _proc_time_std.append(proc_time_std[N_FS][_f])
+        _proc_time = np.array(_proc_time)
+        _proc_time_std = np.array(_proc_time_std)
+
+        plt.loglog(N_FS_vals, _proc_time, label=_f, marker=markers[i])
+        ax = plt.gca()
+        ax.fill_between(
+            N_FS_vals,
+            (_proc_time - n_std * _proc_time_std),
+            (_proc_time + n_std * _proc_time_std),
+            alpha=0.2,
+        )
+
+    plt.legend()
```

Comment:
It's bad style to rely on the implicit fig/ax combo to construct matplotlib figures. (This is the default MATLAB convention, reason why it is so widespread.)
Prefer instead an explicit construction (see previous comment), then call methods of the corresponding `Axes` object (`ax`).
```
ax.legend()
ax.set_title(...)
ax.set_xlabel(...)
ax.set_ylabel(...)
ax.set_xticks(...)
fig.tight_layout()
fig.savefig(fname, dpi=300)
```